docs/core/extensions/sslstream-best-practices.md

```diff
@@ -0,0 +1,142 @@
+
+The alternative solution is to specify custom trusted root certificates via an <xref:System.Security.Cryptography.X509Certificates.X509ChainPolicy>. The example below specifies custom trust list which will be used instead of the system trust list during validation.
+
+```csharp
+SslClientAuthenticationOptions clientOptions = new();
+
+clientOptions.CertificateChainPolicy = new X509ChainPolicy()
+{
+    TrustMode = X509ChainTrustMode.CustomRootTrust,
+    CustomTrustStore = {
+        customIssuerCert
+    }
+};
+```
+
+Note that clients configured with the policy above would accept *only* certificates trusted by `customIssuerCert`.
+
+### Ignoring specific validation errors
+
+Consider an IoT device without a persistent clock. After powering on, the clock of the device would start many years in the past and, therefore, all certificates would be considered "not yet valid". The following code shows a validation callback implementation which ignores validity period violations.
+
+```csharp
+static bool CustomCertificateValidationCallback(
+    object sender,
+    X509Certificate? certificate,
+    X509Chain? chain,
+    SslPolicyErrors sslPolicyErrors)
+{
+    // Anything that would have been accepted by default is OK
+    if (sslPolicyErrors == SslPolicyErrors.None)
+    {
+        return true;
+    }
+    
+    // If there is something wrong other than a chain processing error, don't trust it.
+    if (sslPolicyErrors != SslPolicyErrors.RemoteCertificateChainErrors)
+    {
+        return false;
+    }
+    
+    Debug.Assert(chain is not null);
+    
+    foreach (X509ChainStatus status in chain.ChainStatus)
+    {
+        // If an error other than `NotTimeValid` (or `NoError`) is present, don't trust it.
+        if ((status.Status & ~X509ChainStatusFlags.NotTimeValid) != X509ChainStatusFlags.NoError)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+```
+
+### Certificate pinning
+
+Another situation where custom certificate validation is necessary is when clients expect servers to use a specific certificate, or a certificate from a small set of known certificates. This practice is known as [certificate pinning](https://owasp.org/www-community/controls/Certificate_and_Public_Key_Pinning). The following code snippet shows a validation callback which checks that the server presents a certificate with a specific known public key.
+
+```csharp
+static bool CustomCertificateValidationCallback(
+    object sender,
+    X509Certificate? certificate,
+    X509Chain? chain,
+    SslPolicyErrors sslPolicyErrors)
+{
+    // If there is something wrong other than a chain processing error, don't trust it.
+    if (sslPolicyErrors != SslPolicyErrors.RemoteCertificateChainErrors)
+    {
+        return false;
+    }
```

Comment:
This block would make sense if the caller expects the certificate to not be trusted.  But if it was trusted, why would that make the pinning fail?

In this particular scenario, it'd be something like `if ((sslPolicyErrors & ~SslPolicyErrors.RemoteCertificateChainErrors) != 0)` (ignoring the chain errors bit, fail if any (other) bit is set).... which only allows None or RemoteCertificateChainErrors through.